src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/sentencing/SentencingService.kt

```diff
@@ -1019,59 +1022,112 @@ class SentencingService(
+    bookingIds.forEach { bookingId ->
+      storedProcedureRepository.imprisonmentStatusUpdate(
+        bookingId = bookingId,
+        changeType = ImprisonmentStatusChangeType.UPDATE_SENTENCE.name,
+      )
+    }
+    telemetryClient.trackEvent(
+      "recall-sentences-updated",
+      mapOf(
+        "bookingId" to bookingIds.joinToString { it.toString() },
+        "sentenceSequences" to request.sentences.map { it.sentenceId.sentenceSequence }.joinToString { it.toString() },
+        "offenderNo" to offenderNo,
+      ),
+      null,
+    )
+  }
+
+  fun replaceRecallSentences(offenderNo: String, request: DeleteRecallRequest) {
+    val bookingIds = request.sentences.map { it.sentenceId.offenderBookingId }.toSet()
 
-    // Create or update OffenderFixedTermRecall records if requested
-    if (request.returnToCustody != null) {
-      val enteredByStaff = findStaffByUsername(request.returnToCustody.enteredByStaffUsername)
+    request.sentences.updateSentences()
+    bookingIds.forEach { bookingId ->
+      findOffenderBooking(bookingId).fixedTermRecall = null
+      offenderFixedTermRecallRepository.deleteById(bookingId)
+      storedProcedureRepository.imprisonmentStatusUpdate(
+        bookingId = bookingId,
+        changeType = ImprisonmentStatusChangeType.UPDATE_SENTENCE.name,
+      )
+    }
+    telemetryClient.trackEvent(
+      "recall-sentences-replaced",
+      mapOf(
+        "bookingId" to bookingIds.joinToString { it.toString() },
+        "sentenceSequences" to request.sentences.map { it.sentenceId.sentenceSequence }.joinToString { it.toString() },
+        "offenderNo" to offenderNo,
+      ),
+      null,
+    )
+  }
+
+  private fun ReturnToCustodyRequest?.createOrUpdateBooking(bookingIds: Set<Long>) {
+    if (this != null) {
+      val enteredByStaff = findStaffByUsername(this.enteredByStaffUsername)
       bookingIds.forEach { bookingId ->
         with(findOffenderBooking(bookingId)) {
           if (fixedTermRecall == null) {
             fixedTermRecall = OffenderFixedTermRecall(
-              returnToCustodyDate = request.returnToCustody.returnToCustodyDate,
+              returnToCustodyDate = this@createOrUpdateBooking.returnToCustodyDate,
               staff = enteredByStaff,
-              recallLength = request.returnToCustody.recallLength.toLong(),
+              recallLength = this@createOrUpdateBooking.recallLength.toLong(),
               offenderBooking = this,
             )
           } else {
             with(fixedTermRecall!!) {
-              returnToCustodyDate = request.returnToCustody.returnToCustodyDate
+              returnToCustodyDate = this@createOrUpdateBooking.returnToCustodyDate
               staff = enteredByStaff
-              recallLength = request.returnToCustody.recallLength.toLong()
+              recallLength = this@createOrUpdateBooking.recallLength.toLong()
             }
           }
         }
       }
+    } else {
+      bookingIds.forEach { bookingId ->
```

Comment:
Interesting I had to "delete" both ends of this one-to-one which makes think perhaps I have the mapping wrong 🤔